torch/distributed/_shard/sharded_tensor/_ops/chunk.py

```diff
@@ -0,0 +1,63 @@
+import torch
+from torch.distributed._shard.sharded_tensor import (
+    sharded_op_impl,
+    ShardedTensor,
+)
+from torch.distributed._shard.sharding_spec import ChunkShardingSpec
+
+
+def register_chunk_op(op):
+    @sharded_op_impl(op)
+    def sharded_chunk(types, args=(), kwargs=None, pg=None):
```

Comment:
why the `chunk` op could't use `_register_sharded_op_on_local_tensor`? I feel we could use this and just need to add `extra_check` condition to ensure it's not chunking the sharding dim?

Reply:
The return value here is different. Chunk return a list of sharded tensors.